bundles/org.openhab.core.ui/src/main/java/org/openhab/core/ui/internal/proxy/ProxyServletService.java

```diff
@@ -74,68 +79,58 @@
  * @author Kai Kreuzer - Initial contribution
  * @author John Cocula - added optional Image/Video item= support; refactored to allow use of later spec servlet
  */
-@Component(immediate = true, property = { "service.pid=org.openhab.core.proxy" })
+@NonNullByDefault
+@Component(immediate = true, property = { "service.pid=org.openhab.proxy" })
 public class ProxyServletService extends HttpServlet {
 
     /** the alias for this servlet */
     public static final String PROXY_ALIAS = "proxy";
 
+    private static final long serialVersionUID = -4716754591953017793L;
     private static final String CONFIG_MAX_THREADS = "maxThreads";
     private static final int DEFAULT_MAX_THREADS = 8;
     public static final String ATTR_URI = ProxyServletService.class.getName() + ".URI";
     public static final String ATTR_SERVLET_EXCEPTION = ProxyServletService.class.getName() + ".ProxyServletException";
 
     private final Logger logger = LoggerFactory.getLogger(ProxyServletService.class);
 
-    private static final long serialVersionUID = -4716754591953017793L;
-
-    private Servlet impl;
+    private @Nullable Servlet impl;
 
-    protected HttpService httpService;
-    protected ItemUIRegistry itemUIRegistry;
-    protected ModelRepository modelRepository;
+    protected final HttpService httpService;
+    protected final ItemUIRegistry itemUIRegistry;
+    protected final List<SitemapProvider> sitemapProviders = new ArrayList<>();
```

Comment:
Should we use a `CopyOnWriteArrayList` here to avoid concurrency issues?

Reply:
I'll defer to @kaikreuzer in regards to the concurrency possibility as he would know the upstream way better than myself.   

The issues this covers : 
https://github.com/openhab/openhab-webui/issues/763
https://github.com/openhab/openhab-core/issues/2154
https://github.com/openhab/openhab-webui/issues/745

This is a 2 part fix, the second part if here I think awaiting the merge of this one : 
https://github.com/openhab/openhab-webui/pull/860

Reply:
I've addressed the comment and also added the fixed issues to the PR description.